contracts/LifCrowdsale.sol

```diff
@@ -260,11 +267,52 @@ contract LifCrowdsale is Ownable, Pausable {
       newMVM.fund.value(mmFundBalance)();
       newMVM.transferOwnership(foundationWallet);
 
-      MVM = address(newMVM);
+      MVM = newMVM;
 
     }
   }
 
+  // distribute the rest of the tokens among founders and
+  function finishDistribution() internal {
+    // calculate how much tokens will the founders, foundation and advisors will receive
+    uint256 foundersTokens = token.totalSupply().mul(128).div(1000);
+    uint256 foundationTokens = token.totalSupply().mul(50).div(1000);
+    uint256 advisorsTokens = token.totalSupply().mul(72).div(1000);
+    token.mint(address(this), foundersTokens.add(foundationTokens).add(advisorsTokens));
```

Comment:
F*ck, if we made the vested payment to be able to receive tokens simply via `transfer`, we wouldn't need to mint to then call to `fund`... maybe too late now but I think it would be simpler that way

Reply:
Its almost same because we still have to call the `fund` function no matter what, if we mint the tokens after the payments are created we can avoid doing the approve. If you want I can change it nico, but I still prefer to doing this this way.

Reply:
we have to call `fund` no matter what in the current implementation of `VestedPayment`. If we made `VestedPayment` to not have a `fund` function, we wouldn't have to call it :). One way to do that is to make `VestedPayment` to calculate the percentage of funds that can be distributed on current block, and calculate that amount from the `balanceOf(address(this))`, something like that. But I understand we don't have much time to change it to work that way.

Reply:
That can work, but what happens if the contract receives more tokens in the middle of the schedule, on the next period you will be able to claim more tokens, and after the last period you will claim all of the remaining funds?

Reply:
Exactly. I can't imagine why the contract would receive more tokens, except if someone wants to donate some tokens to the beneficiary of the vested payment?

Reply:
Maybe, we didnt even started and we are receiving donations on the presale wallet, so.. if that happens from an address like an exchange, thank you very much and we took them. If we receive tokens/ethers by mistake and the sender can prove the ownership of the address we can return them after we claim it, if we want.